### ISSUE TITLE
Move client-side-only toast messages server-side for consistency

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#5169]
**Describe the bug**
There are a few toast messages which are created client-side and don't follow the pattern of how we create most of our toast messages.  The pattern is to use `ToastMessage.java `.  These one-off toasts, therefore, don't reflect general changes we may need to make to our toast messaging and could not be localized if we wanted to localize them.

**To Reproduce**
See any .ts files that call `ToastController.showToastMessage` directly.  See below screenshot for usages.  The usages are in the admin predicates and in copying program links.
![image](https://github.com/civiform/civiform/assets/134556723/b6132072-7522-4ad7-a377-40b3c952b092)

**Expected behavior**
All new toast messages are created in a consistent way, by going through the `ToastMessage.java` factory methods.

**Additional context**
Fixing this bug will allow the "Error: " prefix to be added to these toasts as well, which is needed for accessibility.  See [#3286](https://github.com/civiform/civiform/issues/3286).

**Comment by bion** [comment]
@melanie-exygy please clarify a11y implications of this issue

**Comment by melanie-exygy** [comment]
@bion Copied the a11y context in the bug description as well, for clarity.  It's true that the a11y improvements would be for admins, which isn't required, but those improvements would just be a bonus in fixing the bug.

**Comment by shreyachatterjee00** [comment]
@melanie-exygy is this still relevant?

**Comment by melanie-exygy** [comment]
> [@melanie-exygy](https://github.com/melanie-exygy) is this still relevant?

Yes, unfortunately.  However, once the form logic feature is released, we won't be using the toast message for predicate configuration, so that will take care of that one.  

As for the toast message when the admin clicks on "Share link" in the admin program dashboard, I suspect that Jia @myfriendgu has plans to replace that as part of the Admin UX re-design.  So, we should re-visit this issue once the admin re-design has been released.
